### DISCUSSION TITLE
Connect to remote DuckDB instance

### DISCUSSION THREAD

**Discussion by czabriskie** [duckdb/duckdb#5753] [Q&A]
I'm assuming DuckDB is the same as SQLite and meant to be embedded in the software instead of its own standing database. However, I still wanted to ask if it is possible/recommended to connect to a remote duckDB instance? I have one on an EC2 instance I was hoping to connect to locally and query. I don't see a way to do that with a ```duckdb.connection``` object though.

**Comment by Mause** [comment]
@jwills, one of our users, has been experimenting with this recently: https://github.com/jwills/buenavista

**Reply by jwills** [reply]
Hey @czabriskie, guilty-as-charged, although I _just_ pushed a [commit](https://github.com/jwills/buenavista/commit/5276d6137624637e14cc8c0f776f4a78b1203f00) to only allow connections from localhost until I implement the postgres password protocols for authentication. 😅 

As @Mause said, I'm just experimenting with this right now, but I'm planning to work on it a lot over the next several months, there are lots of ideas I would like to try out. Happy to keep in touch!

**Comment by vivianTan777** [comment]
it the feature "connect to a remote duckDB instance" now supported?

**Reply by Mause** [reply]
You can read a remote database file in read only mode via the ATTACH statement?

**Reply by Miljar** [reply]
@Mause I tried this just now, on duckdb 0.10.1, without success:

```-- Config S3 secrets
CREATE SECRET secret1 (
    TYPE S3,
    KEY_ID 'mykey',
    SECRET 'mysecret',
    REGION 'eu-west-1',
    SESSION_TOKEN 'mysessiontoken'
);

-- ATTACH FROM S3
ATTACH 's3://mybucket-eu-west-1/my_temp_db.duckdb' AS interim (READ_ONLY);
```

Which produced the error

```duckdb.duckdb.CatalogException: Catalog Error: Cannot open database "s3://mybucket-eu-west-1/my_temp_db.duckdb" in read-only mode: database does not exist```

But the file does exist and I copied the S3-URL from the S3 console

**Reply by Miljar** [reply]
Sorry, just found this https://github.com/duckdb/duckdb/issues/10409
